### PR TITLE
feat(edgeless): support collapsible note with new style

### LINKED ISSUE #5234: Creating a new Note quickly should result in an empty Note
Currently in the Canary version, clicking the "+" icon on the Note Block to create a new Note quickly carries the content from the previous Note. This should be separate from the effect of creating a copy.

CleanShot 2023-11-07 at 14 51 06@2x

### LINKED ISSUE #5283: Edgeless notes upgraded
[Below are design updates for notes](https://www.figma.com/file/ry2HHWu5XafpuXBD2XOvKd/%F0%9F%9B%96-2023.S4?type=design&node-id=586-16009&mode=design)
Notes UI upgraded:
1. Notes shadow style added
2. Notes color upgraded
3. Notes border style added
4. Notes radius styles added
5. Notes toolbar upgraded

Notes function upgraded:
1. Resize notes
2. Quick create blank notes

### LINKED ISSUE #5387: Note hover flicker when resizing
https://github.com/toeverything/blocksuite/assets/58546692/2daef142-8021-4daf-b657-7b4418a138a2

I think it's better to not have hover effect when resizing.

### PR DESCRIPTION
**Pull request by regischen**
close #5283 
close #5234
close #5387 

Newly added note model fields:

```ts
enum StrokeStyle {
  Solid = 'solid',
  Dashed = 'dash',
  None = 'none',
}

edgeless: {
  style: {
      borderRadius: number,
      borderSize: number,
      borderStyle: StrokeStyle,
      shadowType: string,
    }
    collapse: boolean,
    collapsedHeight: number,
  }
}
```

```ts
const NOTE_SHADOWS = [
  '',
  '--affine-note-shadow-box',
  '--affine-note-shadow-sticker',
  '--affine-note-shadow-paper',
  '--affine-note-shadow-float',
  `--affine-note-shadow-film`,
]

edgeless: {
  style: {
    borderRadius: 8,
    borderSize: 4,
    borderStyle: 'solid',
    shadowType: NOTE_SHADOWS[1],
  }
}
```
In newly added note model, it doesn't have `collapse` and `collapsedHeight` fields.

### CODE REVIEW

File: packages/blocks/src/page-block/edgeless/components/panel/note-shadow-panel.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import { css, html, LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { styleMap } from 'lit/directives/style-map.js';
+
+import {
+  NoteNoShadowIcon,
+  NoteShadowSampleIcon,
+} from '../../../../_common/icons/edgeless.js';
+import { NOTE_SHADOWS } from '../../../../note-block/note-model.js';
```
Comment (doodlewind) [inline packages/blocks/src/page-block/edgeless/components/panel/note-shadow-panel.ts:13]: Don't cross referencing blocks, add const in common modules instead.

File: packages/blocks/src/page-block/edgeless/components/panel/note-shadow-panel.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+const TOOLBAR_SHADOWS = [
+  '',
+  '0px 0.2px 4.8px 0px rgba(66, 65, 73, 0.2), 0px 0px 1.6px 0px rgba(66, 65, 73, 0.2)',
+  '0px 9.6px 10.4px -4px rgba(66, 65, 73, 0.07), 0px 10.4px 7.2px -8px rgba(66, 65, 73, 0.22)',
+  '0px 1.2px 2.4px 4.8px rgba(66, 65, 73, 0.16), 0px 0px 0px 4px rgba(255, 255, 255, 1)',
+  '0px 5.2px 12px 0px rgba(66, 65, 73, 0.13), 0px 0px 0.4px 1px rgba(0, 0, 0, 0.06)',
+  '0px 0px 0px 1.4px rgba(0, 0, 0, 1), 2.4px 2.4px 0px 1px rgba(0, 0, 0, 1)',
+];
+
+const TOOTIPS = [
```
Comment (doodlewind) [inline packages/blocks/src/page-block/edgeless/components/panel/note-shadow-panel.ts:24]: Typo

File: packages/blocks/src/page-block/edgeless/components/utils.ts [hunk 3]
```diff
@@ -66,7 +66,8 @@ export function createButtonPopper(
   popperElement: HTMLElement,
   stateUpdated: (state: { display: 'show' | 'hidden' }) => void = () => {
     /** DEFAULT EMPTY FUNCTION */
-  }
+  },
+  clickAway = true
```
Comment (doodlewind) [inline packages/blocks/src/page-block/edgeless/components/utils.ts:70]: The mixing of `boolean` and `Disposable` typing is unacceptable.

File: packages/blocks/src/note-block/note-model.ts [hunk 4]
```diff
@@ -34,6 +44,14 @@ export const NoteBlockSchema = defineBlockSchema({
     background: DEFAULT_NOTE_COLOR,
     index: 'a0',
     hidden: false,
+    edgeless: {
+      style: {
+        borderRadius: 8,
+        borderSize: 4,
+        borderStyle: 'solid',
+        shadowStyle: NOTE_SHADOWS[1],
```
Comment (doodlewind) [inline packages/blocks/src/note-block/note-model.ts:52]: `shadowStyle` -> `shadowType`

File: tests/edgeless/note.spec.ts [hunk 5]
```diff
@@ -124,6 +123,52 @@ test('resize note in edgeless mode', async ({ page }) => {
...
+  await page.mouse.click(50, 50);
+  await selectNoteInEdgeless(page, ids.noteId);
+
+  const initRect = await getNoteRect(page, ids);
+  const bottomRightResize = page.locator(
+    '.handle[aria-label="bottom-right"] .resize'
+  );
+  const box = await bottomRightResize.boundingBox();
+
+  if (box === null) throw new Error();
```
Comment (doodlewind) [inline tests/edgeless/note.spec.ts:145]: Why not `assertExists`?

File: tests/basic.spec.ts [hunk 6]
```diff
@@ -512,6 +512,16 @@ test(scoped`automatic identify url text`, async ({ page }) => {
 <affine:page>
   <affine:note
     prop:background="--affine-background-secondary-color"
+    prop:edgeless={
```
Comment (doodlewind) [inline tests/basic.spec.ts:515]: This significantly bloats almost all test cases, which is very much a code smell. @Saul-Mirone any idea avoiding the duplicate here?

**Review by doodlewind — CHANGES_REQUESTED** [review]
Thank you for your contribution to the project. I appreciate the effort you've put into this. However, there are some concerns regarding the handling of potential technical debt that we need to address. It's important for the health and sustainability of the project that we proactively discuss and manage technical debt. I recommend we have a more thorough discussion about these aspects before proceeding to the code review stage. This will help us ensure the high quality and maintainability of our codebase. Let's work together to find the best approach to tackle these challenges. Your continued dedication and improvement are vital to our project's success.

(Generated from https://chat.openai.com/share/2565741f-0de3-4da9-a037-07accf96ddc4)

**Comment by regischen** [comment]
> Thank you for your contribution to the project. I appreciate the effort you've put into this. However, there are some concerns regarding the handling of potential technical debt that we need to address. It's important for the health and sustainability of the project that we proactively discuss and manage technical debt. I recommend we have a more thorough discussion about these aspects before proceeding to the code review stage. This will help us ensure the high quality and maintainability of our codebase. Let's work together to find the best approach to tackle these challenges. Your continued dedication and improvement are vital to our project's success.

Thank you for your acknowledgment. Indeed, I should be more proactive in identifying issues that are worth discussing before the code review stage. This is to ensure the high quality and maintainability of our codebase.

**Review by Svaney-ssman — CHANGES_REQUESTED** [review]
Please hide the styles buttons when note[ in hidden mode](https://www.figma.com/file/ry2HHWu5XafpuXBD2XOvKd/%F0%9F%9B%96-2023.S4?type=design&node-id=587-21109&mode=design&t=gL09M2r3oZUR1TqM-4)
image

**Review by doodlewind — APPROVED** [review]
About 1000 lines added in this PR relates to the default props of note block. We should mitigate this in test infra afterwards.

**Review by Svaney-ssman — APPROVED** [review]
good job👏

**Pull request merged**